packages/dds/tree/src/test/utils.ts

```diff
@@ -38,14 +38,45 @@ import { brand, makeArray } from "../util";
 
 // Testing utilities
 
-export function deepFreeze<T>(object: T): void {
-    // Retrieve the property names defined on object
-    const propNames: (keyof T)[] = Object.getOwnPropertyNames(object) as (keyof T)[];
-    // Freeze properties before freezing self
-    for (const name of propNames) {
-        const value = object[name];
-        if (typeof value === "object") {
-            deepFreeze(value);
+const frozenMethod = () => {
+    assert.fail("Object is frozen");
+};
+
+/**
+ * Recursively freezes the given object.
+ * @param object - The object to freeze.
+ * @param freezeMethods - Whether to freeze mutation methods on `Set` and `Map` instances.
+ * Passing `true` mutates the `Set` and `Map` instances encountered in the recursion.
```

Comment:
I think we give a little more detail on the contract of `deepFreeze` with respect to Map and Set depending on the value of `freezeMethods`, as well as briefly explaining why Map and Set are handled specially.

Reply:
What is else is there to say abouf the contract of deepFreeze with respect to Map and Set depending on the value of freezeMethods?

Reply:
Something like "Note: Calling `Object.freeze` on a Set or Map does not prevent it from being mutated. If `freezeMethods` is true, we remove the mutating methods from the Set or Map to ensure that its state cannot be changed. Regardless of the value of `freezeMethods`, `deepFreeze` will freeze the keys and values of a Map or Set."

I don't understand the motivation for the mode where we don't freeze the methods, so maybe the comment should hint at that.